packages/firestore/src/core/firestore_client.ts

```diff
@@ -474,6 +492,7 @@ export class FirestoreClient {
   shutdown(options?: {
     purgePersistenceWithDataLoss?: boolean;
   }): Promise<void> {
+    this.checkClientRunning();
```

Comment:
FWIW- It's tempting to make shutdown() idempotent (i.e. do nothing if it's already been called).  I think that might also mean you wouldn't need to add the `withTestDbNoShutdown()` helper...  WDYT?